src/Tokenizer.js

```diff
@@ -457,17 +457,33 @@ module.exports = class Tokenizer {
   link(src) {
     const cap = this.rules.inline.link.exec(src);
     if (cap) {
-      const lastParenIndex = findClosingBracket(cap[2], '()');
-      if (lastParenIndex > -1) {
-        const start = cap[0].indexOf('!') === 0 ? 5 : 4;
-        const linkLen = start + cap[1].length + lastParenIndex;
-        cap[2] = cap[2].substring(0, lastParenIndex);
-        cap[0] = cap[0].substring(0, linkLen).trim();
-        cap[3] = '';
+      const trimmedUrl = cap[2].trim();
+      if (!this.options.pedantic && trimmedUrl.startsWith('<')) {
+        // commonmark requires matching angle brackets
+        if (!trimmedUrl.endsWith('>')) {
+          return;
+        }
+
+        // ending angle bracket cannot be escaped
+        const rtrimSlash = rtrim(trimmedUrl.slice(0, -1), '\\');
+        if (!((trimmedUrl.length - rtrimSlash.length) & 1)) {
```

Comment:
Is this intended to be the bitwise AND?

Reply:
yes. It is checking if the number of slashes removed is odd.

Reply:
I see. 

Minor nit pick: Ideally, the `!` could be removed and the XOR operator could be used instead of AND.

Or maybe just modulo/remainder 2 which is what I would expect.

Reply:
I will do some benchmark testing for the modulo operator

Reply:
I didn't see any difference between using modulo or xor so I went with modulo.

Reply:
You can replace the `(trimmedUrl.length - rtrimSlash.length) % 2 === 0)` with `(!(trimmedUrl.length - rtrimSlash.length) % 2))` which seems to be a *bit* faster: https://jsbench.me/lwkigdfqus/1